### PR TITLE
Upgrade `bootstrap.sh`

### PR DESCRIPTION
**Pull request by babsey**

### CODE REVIEW

**Review by sontheimer — COMMENTED** [review]
I already did some changes to the bootstrap that include some cleanup and those python packages. 
I will add that to the PR

**Comment by sontheimer** [comment]
I made a cleanup to make it more consistent and readable.
Please note the new 'order' and numbering of the installation process, e.g. moved insite up to step 4.

please check the NEST DESKTOP and INSITE installation.

please have a quick look at the changes to NEST installation

**Comment by mfahdaz** [comment]
I would suggest to change the path to plan XML file when generating the ` run_on_loacl.sh` so it runs the Cosim_NestDesktop_Insite usecase. For example in line 218 instead of

`--action-plan \${CO_SIM_MODULES_ROOT_PATH}/EBRAINS_WorkflowConfigurations/usecase/local/plans/cosim_alpha_brunel_local.xml`

it should be

`--action-plan \${CO_SIM_USE_CASE_ROOT_PATH}/userland/configs/local/plans/cosim_alpha_brunel_local.xml`

File: installation/local/bootstrap.sh [hunk 1]
```diff
@@ -65,45 +66,62 @@ echo "1" | sudo update-alternatives --config mpi 1>/dev/null 2>&1 # --> choosing
...
+# STEP 5 - Install Insite
+#
+sudo apt-get install -y libssl-dev
+mkdir -p $CO_SIM_INSITE
+cd $CO_SIM_INSITE
+git clone --recurse-submodules https://github.com/VRGroupRWTH/insite.git
+
+cd insite
+
+mkdir build_access_node
```
Comment (marcelkrueger) [inline installation/local/bootstrap.sh:92]: We might want to have this as a VAR also?

**Review by babsey — COMMENTED** [review]
For NEST Desktop installation, it looks fine.

File: installation/local/bootstrap.sh [hunk 2]
```diff
@@ -65,45 +66,62 @@ echo "1" | sudo update-alternatives --config mpi 1>/dev/null 2>&1 # --> choosing
...
+#
+# STEP 5 - Install Insite
+#
+sudo apt-get install -y libssl-dev
+mkdir -p $CO_SIM_INSITE
+cd $CO_SIM_INSITE
+git clone --recurse-submodules https://github.com/VRGroupRWTH/insite.git
+
+cd insite
+
```
Comment (marcelkrueger) [inline installation/local/bootstrap.sh:91]:
```suggestion
```

File: installation/local/bootstrap.sh [hunk 3]
```diff
@@ -65,45 +66,62 @@ echo "1" | sudo update-alternatives --config mpi 1>/dev/null 2>&1 # --> choosing
...
+mkdir build_access_node
+cd build_access_node
```
Comment (marcelkrueger) [inline installation/local/bootstrap.sh:93]: We might want to either always have:
```
mkdir xyz
cd xzy
```
or
```
mkdir xyz; cd xyz 
```
just to be consistent. 
c.f. Line 120

File: installation/local/bootstrap.sh [hunk 4]
```diff
@@ -113,12 +131,12 @@ cmake \
     -DPYTHON_INCLUDE_DIR=/usr/include/python3.10 \
     -DPYTHON_LIBRARY=/usr/lib/x86_64-linux-gnu/libpython3.10.so
 
-make -j 3
+make -j 4
```
Comment (marcelkrueger) [inline installation/local/bootstrap.sh:134]: Maybe this should be a variable as well?
You bumped 3 to 4 here but for Insite it is still 3.
Is this on purpose ?

**Comment by sontheimer** [comment]
perfect. If Marcel does not see any problems with INSITE we can merge.

I added the change to the correct action-xml as suggested by Fahad.

I will test and review the automatic script generation then on the VM.

File: installation/local/bootstrap.sh (hunk 4 above)
Comment (sontheimer) [reply installation/local/bootstrap.sh:134]: I will change it. overlooked.
If everything else is fine, we merge

**Review by mfahdaz — APPROVED** [review]

**Pull request merged**